### PR TITLE
[Merged by Bors] - Refresh vm docs and fix bytecode trace output

### PR DESCRIPTION
**Pull request by raskad**
It changes the following:

- Refreshes the vm and debugging docs to represent the current state
- Fix some bytecode trace output
- Rename a field in the `CodeBlock`

### CODE REVIEW

**Review by HalidOdat — APPROVED** [review]
Looks good to me! :)

File: docs/debugging.md [hunk 1]
```diff
@@ -59,23 +35,20 @@ or with interactive mode (REPL):
 cargo run -- --dump-ast # AST dump format is Debug by default.
 ```
 
-These methods will print out the entire parse tree.
+## Bytecode generation and Execution
+
+Once the AST has been generated boa will compile it into bytecode.
+The bytecode is then executed by the vm.
+You can print the bytecode and the executed instructions with the
+flag command-line flag `--trace`.
```
Comment (RageKnify) [inline docs/debugging.md:43]: > with the ~~flag~~ command-line flag

**Review by jedel1043 — APPROVED** [review]
Nice!

**Review by Razican — APPROVED** [review]

**Review by jasonwilliams — APPROVED** [review]

**Review by RageKnify — APPROVED** [review]

**Pull request closed without merging**